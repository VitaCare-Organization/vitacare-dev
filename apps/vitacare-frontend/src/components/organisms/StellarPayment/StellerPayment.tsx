import React from "react";

const StellerPayment = () => {
  const transactions = [
    {
      date: "February 15, 2025",
      recipient: "City General Hospital",
      amount: "-$85.00",
      tx: "TX: STLR_AFRD",
    },
    {
      date: "February 1, 2025",
      recipient: "Bright Smile Dental",
      amount: "-$150.00",
      tx: "TX: STLR_BASIC",
    },
    {
      date: "January 25, 2025",
      recipient: "From Bank Account",
      amount: "+$500.00",
      tx: "TX: STLR_78SE",
    },
  ];
  return (
    <div className="border border-gray-400 rounded-2xl mt-6 overflow-hidden">
      <div className=" p-6  ">
        <h2 className="text-2xl font-semibold text-gray-800 mb-4">
          Stellar Blockchain Wallet
          <br />
          <span className="text-sm text-gray-500">
            Manage your Stellar wallet for healthcare payments
          </span>
        </h2>
        <div className="flex justify-between items-center border border-gray-300 p-4 shadow rounded ">
          <div>
            <p className="text-base text-gray-600 mb-2 ">Wallet Status</p>
            <p className="text-green-500 bg-green-100 px-2 py-1 rounded-2xl inline-flex items-center">
              <svg
                className="w-4 h-4 mr-1"
                fill="none"
                stroke="currentColor"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth="2"
                  d="M5 13l4 4L19 7"
                />
              </svg>
              Blockchain verified.
            </p>
            <span>Connected to Stellar Network</span>
          </div>
          <div>
            <p className="text-sm text-gray-600">Wallet Balance</p>
            <p className="text-2xl font-bold text-gray-800">$350.00 USD</p>
          </div>
        </div>
      </div>

      {/* Transaction History */}
      <div className=" p-6">
        <div className="border border-gray-300 rounded-2xl p-4">
          <h2 className="text-lg font-semibold text-gray-800 mb-4">
            Transaction History
          </h2>
          {transactions.map((transaction, index) => (
            <div
              key={index}
              className="flex justify-between items-center py-3  last:border-b-0"
            >
              <div>
                <p className="text-gray-800">{transaction.recipient}</p>
                <p className="text-sm text-gray-500">{transaction.date}</p>
              </div>
              <div className="text-right">
               
                <p
                  className={`font-medium ${
                    transaction.amount.startsWith("-")
                      ? "text-red-500"
                      : "text-green-500"
                  }`}
                >
                  {transaction.amount}
                </p>
                <p className="text-sm text-gray-500">{transaction.tx}</p>
              </div>
            </div>
          ))}
        </div>
        <div className="flex justify-between items-center mt-4">
          <button className="bg-[#0096CC] w-[48%] text-white px-4 py-2 rounded hover:bg-[#558ca0]">
            Add Funds
          </button>
          <button className="text-gray-500 flex items-center justify-center border-gray-300 rounded space-x-1 w-[48%] border md:py-2 px-4 ">
            <span className="text-sm">Export Transactions</span>
            <svg
              className="w-4 h-4"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="2"
                d="M4 16v2a2 2 0 002 2h12a2 2 0 002-2v-2m-4-4l-4 4m0 0l-4-4m4 4V4"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  );
};

export default StellerPayment;
