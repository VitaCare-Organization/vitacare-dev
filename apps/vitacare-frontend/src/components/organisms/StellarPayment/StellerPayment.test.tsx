import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import StellerPayment from "./StellerPayment";

describe("StellerPayment Component", () => {
  it("displays wallet status as blockchain verified", () => {
    render(<StellerPayment />);

    expect(screen.getByText("Wallet Status")).toBeInTheDocument();
    expect(screen.getByText("Blockchain verified.")).toBeInTheDocument();
    expect(screen.getByText("Connected to Stellar Network")).toBeInTheDocument();
  });

  it("shows the wallet balance", () => {
    render(<StellerPayment />);

    expect(screen.getByText("Wallet Balance")).toBeInTheDocument();
    expect(screen.getByText("$350.00 USD")).toBeInTheDocument();
  });

  it("renders transaction history section", () => {
    render(<StellerPayment />);

    expect(screen.getByRole("heading", { name: /transaction history/i })).toBeInTheDocument();
  });

  it("displays all transaction entries", () => {
    render(<StellerPayment />);

    expect(screen.getByText("City General Hospital")).toBeInTheDocument();
    expect(screen.getByText("Bright Smile Dental")).toBeInTheDocument();
    expect(screen.getByText("From Bank Account")).toBeInTheDocument();
  });

  it("shows transaction dates", () => {
    render(<StellerPayment />);

    expect(screen.getByText("February 15, 2025")).toBeInTheDocument();
    expect(screen.getByText("February 1, 2025")).toBeInTheDocument();
    expect(screen.getByText("January 25, 2025")).toBeInTheDocument();
  });

  it("displays transaction amounts with correct colors", () => {
    render(<StellerPayment />);

    const negativeAmounts = screen.getAllByText(/^-\$/);
    const positiveAmount = screen.getByText("+$500.00");

    negativeAmounts.forEach(amount => {
      expect(amount).toHaveClass("text-red-500");
    });

    expect(positiveAmount).toHaveClass("text-green-500");
  });

  it("shows transaction IDs", () => {
    render(<StellerPayment />);

    expect(screen.getByText("TX: STLR_AFRD")).toBeInTheDocument();
    expect(screen.getByText("TX: STLR_BASIC")).toBeInTheDocument();
    expect(screen.getByText("TX: STLR_78SE")).toBeInTheDocument();
  });

  it("renders action buttons", () => {
    render(<StellerPayment />);

    expect(screen.getByRole("button", { name: /add funds/i })).toBeInTheDocument();
    expect(screen.getByRole("button", { name: /export transactions/i })).toBeInTheDocument();
  });

  it("Add Funds button has correct styling", () => {
    render(<StellerPayment />);

    const addFundsButton = screen.getByRole("button", { name: /add funds/i });
    expect(addFundsButton).toHaveClass("bg-[#0096CC] w-[48%] text-white px-4 py-2 rounded hover:bg-[#558ca0]");
  });

  it("Export Transactions button has correct styling", () => {
    render(<StellerPayment />);

    const exportButton = screen.getByRole("button", { name: /export transactions/i });
    expect(exportButton).toHaveClass("text-gray-500 flex items-center justify-center border-gray-300 rounded space-x-1 w-[48%] border md:py-2 px-4");
  });

  it("buttons are clickable", async () => {
    const user = userEvent.setup();
    render(<StellerPayment />);

    const addFundsButton = screen.getByRole("button", { name: /add funds/i });
    const exportButton = screen.getByRole("button", { name: /export transactions/i });

    await user.click(addFundsButton);
    await user.click(exportButton);
  });
}); 