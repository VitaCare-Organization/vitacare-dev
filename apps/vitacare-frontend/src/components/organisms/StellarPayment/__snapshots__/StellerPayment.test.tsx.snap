// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StellerPayment Component matches snapshot 1`] = `
<div
  class="border border-gray-400 rounded-2xl mt-6 overflow-hidden"
>
  <div
    class=" p-6  "
  >
    <h2
      class="text-2xl font-semibold text-gray-800 mb-4"
    >
      Stellar Blockchain Wallet
      <br />
      <span
        class="text-sm text-gray-500"
      >
        Manage your Stellar wallet for healthcare payments
      </span>
    </h2>
    <div
      class="flex justify-between items-center border border-gray-300 p-4 shadow rounded "
    >
      <div>
        <p
          class="text-base text-gray-600 mb-2 "
        >
          Wallet Status
        </p>
        <p
          class="text-green-500 bg-green-100 px-2 py-1 rounded-2xl inline-flex items-center"
        >
          <svg
            class="w-4 h-4 mr-1"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M5 13l4 4L19 7"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
          Blockchain verified.
        </p>
        <span>
          Connected to Stellar Network
        </span>
      </div>
      <div>
        <p
          class="text-sm text-gray-600"
        >
          Wallet Balance
        </p>
        <p
          class="text-2xl font-bold text-gray-800"
        >
          $350.00 USD
        </p>
      </div>
    </div>
  </div>
  <div
    class=" p-6"
  >
    <div
      class="border border-gray-300 rounded-2xl p-4"
    >
      <h2
        class="text-lg font-semibold text-gray-800 mb-4"
      >
        Transaction History
      </h2>
      <div
        class="flex justify-between items-center py-3  last:border-b-0"
      >
        <div>
          <p
            class="text-gray-800"
          >
            City General Hospital
          </p>
          <p
            class="text-sm text-gray-500"
          >
            February 15, 2025
          </p>
        </div>
        <div
          class="text-right"
        >
          <p
            class="font-medium text-red-500"
          >
            -$85.00
          </p>
          <p
            class="text-sm text-gray-500"
          >
            TX: STLR_AFRD
          </p>
        </div>
      </div>
      <div
        class="flex justify-between items-center py-3  last:border-b-0"
      >
        <div>
          <p
            class="text-gray-800"
          >
            Bright Smile Dental
          </p>
          <p
            class="text-sm text-gray-500"
          >
            February 1, 2025
          </p>
        </div>
        <div
          class="text-right"
        >
          <p
            class="font-medium text-red-500"
          >
            -$150.00
          </p>
          <p
            class="text-sm text-gray-500"
          >
            TX: STLR_BASIC
          </p>
        </div>
      </div>
      <div
        class="flex justify-between items-center py-3  last:border-b-0"
      >
        <div>
          <p
            class="text-gray-800"
          >
            From Bank Account
          </p>
          <p
            class="text-sm text-gray-500"
          >
            January 25, 2025
          </p>
        </div>
        <div
          class="text-right"
        >
          <p
            class="font-medium text-green-500"
          >
            +$500.00
          </p>
          <p
            class="text-sm text-gray-500"
          >
            TX: STLR_78SE
          </p>
        </div>
      </div>
    </div>
    <div
      class="flex justify-between items-center mt-4"
    >
      <button
        class="bg-[#0096CC] w-[48%] text-white px-4 py-2 rounded hover:bg-[#558ca0]"
      >
        Add Funds
      </button>
      <button
        class="text-gray-500 flex items-center justify-center border-gray-300 rounded space-x-1 w-[48%] border md:py-2 px-4 "
      >
        <span
          class="text-sm"
        >
          Export Transactions
        </span>
        <svg
          class="w-4 h-4"
          fill="none"
          stroke="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M4 16v2a2 2 0 002 2h12a2 2 0 002-2v-2m-4-4l-4 4m0 0l-4-4m4 4V4"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
